Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -772,35 +788,27 @@ public struct ObjectId: BSONValue, Equatable, CustomStringConvertible, Codable {
         throw bsonEncodingUnsupportedError(value: self, at: to.codingPath)
     }
 
-    /// Initializes an `ObjectId` from an `UnsafePointer<bson_oid_t>` by copying the data
-    /// from it to a `String`
-    internal init(fromPointer oid_t: UnsafePointer<bson_oid_t>) {
-        var str = Data(count: 25)
-        self.oid = str.withUnsafeMutableBytes { (bytes: UnsafeMutablePointer<Int8>) in
-            bson_oid_to_string(oid_t, bytes)
-            return String(cString: bytes)
-        }
-        self.timestamp = UInt32(bson_oid_get_time_t(oid_t))
+    /// Initializes an `ObjectId` from an `UnsafePointer<bson_oid_t>` by copying the underlying `bson_oid_t`.
+    internal init(copying oid_t: UnsafePointer<bson_oid_t>) {
+        self.oid = oid_t.pointee
     }
 
     /// Returns the provided string as a `bson_oid_t`.
     /// - Throws:
     ///   - `UserError.invalidArgumentError` if the parameter string does not correspond to a valid `ObjectId`.
     internal static func toLibBSONType(_ str: String) throws -> bson_oid_t {
```

Comment:
is this method still used anywhere or can we delete it?

Reply:
nope, removed.